src/coreclr/tools/Common/Internal/Runtime/RiscVLoongArch64FpStruct.cs

```diff
@@ -189,6 +207,20 @@ public static FpStructInRegistersInfo GetFpStructInRegistersInfo(TypeDesc td, Ta
                 return new FpStructInRegistersInfo{}; // struct has no floating fields
 
             Debug.Assert(nFields == 1 || nFields == 2);
+            if (nFields == 2 && info.offset1st > info.offset2nd)
+            {
+                // swap fields to match memory order
+                info.flags = (FpStruct)(
+                    ((uint)(info.flags & FloatInt) << (PosIntFloat - PosFloatInt)) |
+                    ((uint)(info.flags & IntFloat) >> (PosIntFloat - PosFloatInt)) |
+                    ((uint)(info.flags & SizeShift1stMask) << (PosSizeShift2nd - PosSizeShift1st)) |
+                    ((uint)(info.flags & SizeShift2ndMask) >> (PosSizeShift2nd - PosSizeShift1st))
+                );
+                (info.offset2nd, info.offset1st) = (info.offset1st, info.offset2nd);
+            }
```

Comment:
Is `nFields == 2` the only case that needs to be handled? What about e.g. 8 byte fields and a double field? Can that combination not be passed in an integer + float register?